diff -uprN a/gio/giomodule.c b/gio/giomodule.c
--- a/gio/giomodule.c	2022-09-25 09:39:14.677169100 +0900
+++ b/gio/giomodule.c	2022-09-25 09:46:52.194591500 +0900
@@ -1128,10 +1128,11 @@ DllMain (HINSTANCE hinstDLL,
 #pragma G_DEFINE_CONSTRUCTOR_PRAGMA_ARGS(gio_win32_ctor)
 #endif
 G_DEFINE_CONSTRUCTOR(gio_win32_ctor)
-
+extern void gobject_init_w32 (void);
 static void
 gio_win32_ctor(void)
 {
+  gobject_init_w32 ();
   gio_win32_appinfo_init (FALSE);
 }
 #else
diff -uprN a/gobject/gtype.c b/gobject/gtype.c
--- a/gobject/gtype.c	2022-09-25 09:39:14.719386400 +0900
+++ b/gobject/gtype.c	2022-09-25 09:44:09.259401400 +0900
@@ -4429,11 +4429,16 @@ g_type_init (void)
 static void
 gobject_init (void)
 {
+  static gboolean gobject_inited;
   const gchar *env_string;
   GTypeInfo info;
   TypeNode *node;
   GType type G_GNUC_UNUSED  /* when compiling with G_DISABLE_ASSERT */;
 
+  if (gobject_inited)
+    return;
+  gobject_inited = TRUE;
+
   /* Ensure GLib is initialized first, see
    * https://bugzilla.gnome.org/show_bug.cgi?id=756139
    */
@@ -4521,6 +4526,12 @@ gobject_init (void)
   _g_signal_init ();
 }
 
+void gobject_init_w32 (void);
+void gobject_init_w32 (void)
+{
+  gobject_init ();
+}
+
 #if defined (G_OS_WIN32) && defined (DLL_EXPORT)
 
 BOOL WINAPI DllMain (HINSTANCE hinstDLL,
